Reference fix in encoding `PrustiBuiltin::Forall`

paaassscccaaalll opened the pull request:
Solves issue where forall annotations could not handle references.  It checks whether the closure captures variables containing references, if it does, it does not put `(make_concrete_s_prusti_pre_item_..._Closure_0(s_Ref_immutable_value(...)))` around it.

I'm aware that right now this is not solved in a very elegant way and we might need to find a better way to solve this and solve the underlying issue.

JonasAlaif commented on prusti-encoder/src/encoders/mir_pure.rs:
```diff
@@ -1030,20 +1030,32 @@ impl<'vir: 'enc, 'enc> Enc<'vir, 'enc> {
                         .map(|qvar| self.vcx.mk_local_ex(qvar.name, qvar.ty)),
                 );
-
+                
```
Make sure to always check the diff of the PR (https://github.com/Aurel300/prusti-dev/pull/71/files) when you make one such that things like this aren't part of it

Aurel300 commented:
Could you add a test case that makes use of this code?

paaassscccaaalll commented:
I now added a test for it. Altough I am not quite sure whether I put it in the right palce. 

It is here now:
`prusti-dev/prusti-tests/tests/verify/pass/quantifiers/method_calls.rs`

Note: It needs the additional axiom from #70 to successfully verify.

Aurel300 commented on prusti-tests/tests/verify/pass/quantifiers/method_calls.rs:
Please place this somewhere in `prusti-tests/tests/v2/pass`, for now.

JonasAlaif committed:
Hack to avoid type errors

style fix

style fix

style fix

added test for PR 71

moved test to v2/pass

removed spaces
